src/systemctl/systemctl.c

```diff
@@ -5723,12 +5721,11 @@ static int add_dependency(int argc, char *argv[], void *userdata) {
 }
 
 static int preset_all(int argc, char *argv[], void *userdata) {
+        UnitFileChange *changes = NULL;
+        unsigned n_changes = 0;
         int r;
 
         if (install_client_side()) {
-                UnitFileChange *changes = NULL;
-                unsigned n_changes = 0;
-
                 r = unit_file_preset_all(arg_scope, arg_runtime, arg_root, arg_preset_mode, arg_force, &changes, &n_changes);
                 unit_file_dump_changes(r, "preset", changes, n_changes, arg_quiet);
                 unit_file_changes_free(changes, n_changes);
```

Comment:
You have to move the unit_file_changes_free() call outside of this block too, if you move the declaration out of it. Otherwise you leak memory.